test/onair/src/systems/test_vehicle_rep.py

```diff
@@ -309,7 +330,36 @@ def test_VehicleRepresentation_update_does_not_raise_IndexError_when_frame_locat
 
     # Assert
     assert e_info.match(fake_non_IndexError_message)
+    
+def test_VehicleRepresentation_update_calls_update_with_given_frame_on_each_knowledge_synthesis_constructs_item(mocker):
+    # Arrange
+    arg_frame = MagicMock()
+    arg_headers = MagicMock()
+    arg_tests = MagicMock()
+
+    cut = VehicleRepresentation.__new__(VehicleRepresentation)
+    cut.__init__(arg_headers, arg_tests) # Init to fill attributes
+    cut.knowledge_synthesis_constructs = []
 
+    num_fake_knowledge_synthesis_constructs = pytest.gen.randint(0, 10) # arbitrary, from 0 to 10
```

Comment:
Doesn't always pass coverage test because if knowledge_systhesis_constructs gets set to 0, the loop here is never run:

def update(self, frame): 
      # Update constructs 

      for construct in self.knowledge_synthesis_constructs: 
          construct.update(frame)

Alan already has a fix coming :)

Reply:
Fixed, but the entire update function was reworked